app/gui/view/graph-editor/src/selection/bounding_box.rs

```diff
@@ -74,6 +76,45 @@ impl BoundingBox {
         self.bottom -= size / 2.0;
         self.top += size / 2.0;
     }
+
+    /// Return the x position of the left boundary.
+    pub fn left(&self) -> f32 {
+        self.left
+    }
+
+    /// Return the x position of the right boundary.
+    pub fn right(&self) -> f32 {
+        self.right
+    }
+
+    /// Return the y position of the top boundary.
+    pub fn top(&self) -> f32 {
+        self.top
+    }
+
+    /// Return the y position of the bottom boundary.
+    pub fn bottom(&self) -> f32 {
+        self.bottom
+    }
```

Comment:
What is better in making function accessors over making the fields public?

Reply:
As we [discussed](https://discord.com/channels/401396655599124480/949323065320415292/949326753753600052), function accessors *"make fields of a type accessible in a way that guards against breaking the type's invariants"* (the invariants established by the object's constructor). Based on your suggestion, I expanded the `BoundingBox` type documentation comment, attempting to highlight the invariants.